src/useFormState.ts

```diff
@@ -106,16 +108,21 @@ export function useFormState<T, I>(opts: UseFormStateOpts<T, I>): ObjectState<T>
         // Don't use canSave() because we don't want to set touched for all the fields
         if (autoSaveRef.current && form.dirty && form.valid && !isAutoSaving) {
           isAutoSaving = "queued";
+          let maybeError: undefined | string;
           // We use setTimeout as a cheap way to wait until the end of the current event listener
           setTimeout(async () => {
             try {
               // We technically don't flip to in-flight until after the call in case the
               // user's autoSave function itself wants to call a .set.
               const promise = autoSaveRef.current!(form);
               isAutoSaving = "in-flight";
+              autoSaveStatusContext.triggerAutoSave();
               await promise;
+            } catch (e) {
+              maybeError = String(e);
```

Comment:
Is this a behavior change, that we need to re-throw `maybeError` inside of the `finally`, b/c as-is we went from throwing e to suppressing e? 

Like at the end of the `finally` an `if (maybeError) throw maybeError`?

Reply:
I thought try/finally was suppressing the error. Happy to be wrong about that.

Reply:
Ah yeah, no, `finally` w/o a `catch` runs the code but then still throws the error. 

I suppose also:

```
try {
  ...
} catch (e) {
  maybeError = String(e);
  throw e;
} finally {
  ...
}
```

Will keep the same semantics.

Reply:
Will `String(e)` be `e.message`?

Reply:
If `e` is an Error, yes. TypeScript types `catch (e)` as `e: unknown`.